src/operators.jl

```diff
@@ -858,3 +855,25 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, v::Vector{U})  where U <: Integer
+    vs = Set(v)
+    v0 = min(vs...)
+    new_vertex_ids = collect(vertices(g))
+    for v in reverse(collect(vs))[1:end-1]
```

Comment:
This will allocate another vector of size `vs`.  You're also implying some sort of ordering to the set (because you're using `reverse`), when sets are unordered.